src/workerd/api/gpu/gpu-shader-module.h

```diff
@@ -4,20 +4,82 @@
 
 #pragma once
 
+#include "gpu-async-runner.h"
+#include "gpu-utils.h"
 #include <webgpu/webgpu_cpp.h>
 #include <workerd/jsg/jsg.h>
 
 namespace workerd::api::gpu {
 
+class GPUCompilationMessage : public jsg::Object {
+public:
+  explicit GPUCompilationMessage(const WGPUCompilationMessage& m)
+      : message(m) {}
+
+  JSG_RESOURCE_TYPE(GPUCompilationMessage) {
+    JSG_READONLY_PROTOTYPE_PROPERTY(message, getMessage);
+    JSG_READONLY_PROTOTYPE_PROPERTY(type, getType);
+    JSG_READONLY_PROTOTYPE_PROPERTY(lineNum, getLineNum);
+    JSG_READONLY_PROTOTYPE_PROPERTY(linePos, getLinePos);
+    JSG_READONLY_PROTOTYPE_PROPERTY(offset, getOffset);
+    JSG_READONLY_PROTOTYPE_PROPERTY(length, getLength);
+  }
+
+private:
+  WGPUCompilationMessage message;
+
+  kj::StringPtr getMessage() { return message.message; }
+  GPUCompilationMessageType getType() {
+    switch (message.type) {
+    case WGPUCompilationMessageType_Error:
+      return kj::str("error");
+    case WGPUCompilationMessageType_Warning:
+      return kj::str("warning");
+    case WGPUCompilationMessageType_Info:
+      return kj::str("info");
+    default:
+      KJ_UNREACHABLE
+    }
+  }
+  uint64_t getLineNum() { return message.lineNum; }
+  uint64_t getLinePos() { return message.linePos; }
+  uint64_t getOffset() { return message.offset; }
+  uint64_t getLength() { return message.length; }
```

Comment:
Just confirming... these properties are expected to be `bigint` in JavaScript? That's what `uint64_t` will be mapped to so just confirming.

Reply:
In the [spec](https://www.w3.org/TR/webgpu/#dom-gpucompilationmessage-linenum) they have the type [unsigned long long](https://webidl.spec.whatwg.org/#idl-unsigned-long-long).

Reply:
Done.